src/main/app.js

```diff
@@ -59,37 +67,83 @@ fluid.defaults("gpii.app", {
             activeSet: null
         }
     },
-    // prerequisites
+    members: {
+        machineId: "@expand:{that}.installID.getMachineID()"
+    },
     components: {
-        psp: {
-            type: "gpii.app.psp",
-            createOnEvent: "onPrerequisitesReady",
+        installID: {
+            type: "gpii.installID"
+        },
+        factsManager: {
+            type: "gpii.app.factsManager",
+            priority: "after:installID"
+        },
+        surveyManager: {
+            type: "gpii.app.surveyManager",
+            priority: "after:factsManager"
+        },
+        dialogManager: {
+            type: "gpii.app.dialogManager",
+            createOnEvent: "onPSPPrerequisitesReady",
+            priority: "after:surveyManager",
```

Comment:
This annotation can be removed since the component's definition includes a dependency on the surveyManager - in addition, this component is createOnEvent and so will definitely be constructed later